src/peft/peft_model.py

```diff
@@ -136,32 +143,49 @@ def save_pretrained(self, save_directory: str, safe_serialization: bool = False,
         os.makedirs(save_directory, exist_ok=True)
         self.create_or_update_model_card(save_directory)
 
-        for adapter_name, peft_config in self.peft_config.items():
-            # save only the trainable weights
-            output_state_dict = get_peft_model_state_dict(
-                self, state_dict=kwargs.get("state_dict", None), adapter_name=adapter_name
-            )
-            output_dir = os.path.join(save_directory, adapter_name) if adapter_name != "default" else save_directory
-            os.makedirs(output_dir, exist_ok=True)
+        if selected_adapters is None:
```

Comment:
I would suggest to move this check above the directory creation. That way, if an error is raised, no useless directories are created.

In theory, we could also do an early return (or raise) if `selected_adapters` is an empty list, not sure if that is a mistake that's likely to happen though.

Reply:
Makes sense!

Reply:
Fixed in [9ae3211](https://github.com/huggingface/peft/pull/705/commits/9ae32111f8cd7ef58cdfb60d9fa7f55a1ad676fe)